src/zcash/JoinSplit.cpp

```diff
@@ -156,48 +157,44 @@ namespace libzcash {
             return GrothProof();
         }
 
-        GrothProof proof;
-
         CDataStream ss1(SER_NETWORK, PROTOCOL_VERSION);
         ss1 << inputs[0].witness.path();
-        std::vector<unsigned char> auth1(ss1.begin(), ss1.end());
+        std::array<unsigned char, 966> auth1;
```

Comment:
Can this be a named constant (or calculated from named constants)?

Reply:
It is a named and calculated constant on the Rust side: https://github.com/zcash/librustzcash/blob/1e39daf6f83c4b62d645e74385d24299dc4414c7/zcash_proofs/src/sprout.rs#L15

The Rust compiler type-checks the Rust API (that uses the named constant) against the Rust side of the `cxx` API, and Clang type-checks the C++ side of the `cxx` API against this type. So I'm fine with this as-is.